xonsh/main.py

```diff
@@ -190,7 +191,8 @@ def premain(argv=None):
                     'completer': False,
                     'login': False,
                     'scriptcache': args.scriptcache,
-                    'cacheall': args.cacheall}
+                    'cacheall': args.cacheall,
+                    'ctx': builtins.__xonsh_ctx__}
```

Comment:
This was needed to make sure that, if `register-completer` is invoked inside the xonshrc, that it actually has access to the functions that are defined there (since `register-completer` looks in `builtins.__xonsh_ctx__` for them).